File: pages/DropdownPage.ts
import { Page, Locator } from '@playwright/test';
import { BasePage } from './BasePage';

/**
 * DropdownPage — Page Object for /dropdown
 * Covers TC-003 (dropdown selection).
 */
export class DropdownPage extends BasePage {
  readonly dropdown: Locator;

  constructor(page: Page) {
    super(page);
    this.dropdown = page.locator('#dropdown');
  }

  async goto(): Promise<void> {
    await this.navigate('/dropdown');
  }

  async selectOption(optionText: string): Promise<void> {
    await this.dropdown.selectOption({ label: optionText });
  }

  async getSelectedOption(): Promise<string> {
    return this.dropdown.evaluate((el: HTMLSelectElement) => {
      return el.options[el.selectedIndex]?.text ?? '';
    });
  }

  async getAllOptions(): Promise<string[]> {
    return this.dropdown.evaluate((el: HTMLSelectElement) => {
      return Array.from(el.options).map((o) => o.text);
    });
  }
}

File: selenium-java-pom/src/main/java/com/ailearning/seleniumpom/pages/DropdownPage.java
package com.ailearning.seleniumpom.pages;

import java.util.List;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.ui.Select;

public class DropdownPage extends BasePage {
    private static final By DROPDOWN = By.id("dropdown");
    private static final By OPTIONS = By.cssSelector("#dropdown option");

    public DropdownPage(WebDriver driver) {
        super(driver);
    }

    public void gotoPage() {
        navigate("/dropdown");
    }

    public void selectOption(String optionText) {
        Select select = new Select(waitForVisible(DROPDOWN));
        select.selectByVisibleText(optionText);
    }

    public String getSelectedOption() {
        Select select = new Select(waitForVisible(DROPDOWN));
        return select.getFirstSelectedOption().getText().trim();
    }

    public List<String> getAllOptions() {
        return waitForAll(OPTIONS).stream()
                .map(WebElement::getText)
                .map(String::trim)
                .toList();
    }
}
